hack/tools/tools.go

```diff
@@ -7,4 +7,5 @@ import (
 	_ "github.com/onsi/ginkgo/ginkgo"
 	_ "golang.org/x/tools/cmd/goimports"
 	_ "k8s.io/code-generator"
+	_ "k8s.io/kubernetes/cmd/preferredimports"
```

Comment:
could you please explain the  reason why we need to add this import? :) I don't know why.

Reply:
Since we can't install it by `go install`, we need to use `go run` to run it, so we need to keep this in the vendor. See `hack/update-import-aliases.sh` for more info.

Reply:
I'm very very hesitant to involve `k8s.io/kubernetes` dependencies as it will make the `go.mod` mess, see all the `replace` instructions at [here](https://github.com/karmada-io/karmada/blob/393769e79ec82d87ead4abc3e23107927351445d/go.mod#L37-L63).

The `replace` instruction also block people's install commands by `go get` or `go install` just like you mentioned in the comments.

By the way,  we always wish to remove the `k8s.io/kubernetes` dependency.

Reply:
Are there any solutions to help people promote Go 1.17 and don't depend on `k8s.io/kubernetes`?

Reply:
Yes, there is another way, but it looks a bit tricky, we can directly run the preferredimports file in the go mod folder, eg: `go run "$GOPATH"/pkg/mod/k8s.io/kubernetes@v1.21.3/cmd/preferredimports/preferredimports.go`

Reply:
That's an option.
I need some time to investigate this issue. Just share my ideas(options):
- Move `preferredimports` from k/k core repo to somewhere, such as `component-helers`.
- Copy `preferredimports` to karmada repo with the copyright notice.

Reply:
> I need some time to investigate this issue.

Any update on this? Moving `preferredimports` to `component-helpers` and `component-base` will not work, both of their `go.mod` have the `replace` instruction, and their `go.mod` is automatically generated, the `replace` instruction may not be easy to adjust.

> * Copy `preferredimports` to karmada repo with the copyright notice.

So maybe this is our best choice since the `preferredimports` is a very stable tool and will not be updated frequently. WDYT?

Reply:
> So maybe this is our best choice since the preferredimports is a very stable tool and will not be updated frequently. WDYT?

So, let's do it.